tests/optional.cpp

```diff
@@ -334,6 +383,20 @@ TEST_CASE("optional::if_unset()", "[optional]")
 
     // result must be set now, and should be 2
     result.if_set([](auto val) { REQUIRE(val == 2); }).if_unset([]() { FAIL("if_unset must not be called on an engaged optional"); });
+
+    SECTION("argument forwarding")
+    {
+        optional o { 1 };
+        auto test_argument = 2;
+
+        // functor must be called if optional is disengaged
+        o = std::nullopt;
+        auto result = o.if_unset([](auto& test_argument) { return 2 + test_argument; }, test_argument);
+        REQUIRE(result == 4);
+
+        // result must be set now, and should be 4
+        result.if_set([](auto val) { REQUIRE(val == 4); }).if_unset([]() { FAIL("if_unset must not be called on an engaged optional"); });
```

Comment:
both of these conditions are already implicitly checked with the `REQUIRE(result == 4)` above